### PR TITLE
Angular - Make TS types  stricter

### PR DESCRIPTION
**Pull request by artem-kurchenko**

### CODE REVIEW

File: Angular/src/app/data.service.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+  AddressRequired?: boolean;
+  StateID?: number;
 }
 
 export class State {
-  ID: number;
-  Name: string;
+  ID!: number;
+  Name?: string;
 }
```
Comment (pomahtri) [inline Angular/src/app/data.service.ts]: seems it should be interface instead of class, and then we could get rid of `!` and `?`

Comment (artem-kurchenko) [reply Angular/src/app/data.service.ts]: Well makes sense, I'll double-check this part.  Initially we used our services code from demos: https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/SimpleArray/Angular/Light/
everywhere)

Comment (pomahtri) [reply Angular/src/app/data.service.ts]: Hm, seems we need to update it too

Comment (artem-kurchenko) [reply Angular/src/app/data.service.ts]: Updated this code and added a [card](https://trello.com/c/EI30zz4Y/3834-wg-demos-ts-update-class-to-interfaces-in-angular-demos) for a squad

**Review by pomahtri — APPROVED** [review]

File: Angular/src/app/app.component.ts [hunk 2]
```diff
@@ -1,45 +1,64 @@
...
-      const rowIndex = gridInstance.getRowIndexByKey(editRowKey);
-      item.visible = gridInstance.cellValue(rowIndex, "AddressRequired");
-    }
+  isHomeAddressGroup(item: GroupItem): boolean {
+    return item && item.itemType === 'group' && item.caption === 'Home Address';
   }
-  onEditorPreparing(e) {
+  customizeItem = (item: Item) => {
+    if (this.isHomeAddressGroup(item)) {
+      const gridInstance: dxDataGrid<Employee, number> = this.dataGrid.instance;
```
Comment (IlyaKhD) [inline Angular/src/app/app.component.ts:41]: **Minor**
Isn't this explicit typing redundant? I guess `this.dataGrid.instance` is already typed.
```suggestion
      const gridInstance = this.dataGrid.instance;
```

File: Angular/src/app/app.component.ts [hunk 3]
```diff
@@ -1,45 +1,64 @@
...
+      const editing = gridInstance.option('editing');
+      const rowIndex = gridInstance.getRowIndexByKey(editing?.editRowKey!);
+      item.visible = gridInstance.cellValue(rowIndex, 'AddressRequired');
+    }
+  };
+  onEditorPreparing(e: EditorPreparingEvent<Employee, number>) {
     if (e.dataField === 'LastName' && e.parentType === 'dataRow') {
-      e.editorOptions.disabled = e.row.data && e.row.data.FirstName === '';
+      e.editorOptions.disabled = e.row?.data && e.row?.data.FirstName === '';
```
Comment (IlyaKhD) [inline Angular/src/app/app.component.ts]: **Minor**
If you're using optional chaining, you can shorten it:
```suggestion
      e.editorOptions.disabled = e.row?.data?.FirstName === '';
```

**Review by IlyaKhD — APPROVED** [review]

File: Angular/src/app/app.component.ts (hunk 3 above)
Comment (artem-kurchenko) [reply Angular/src/app/app.component.ts]: changed

File: Angular/src/app/app.component.ts (hunk 2 above)
Comment (artem-kurchenko) [reply Angular/src/app/app.component.ts:41]: Discussed in PM and created one more [card](https://trello.com/c/fKVYb1yE/3847-ts-support-generics-for-angular-wrappers)

**Pull request merged**